Objects/typeobject.c

```diff
@@ -1674,6 +1674,68 @@ type_set_doc(PyTypeObject *type, PyObject *value, void *context)
     return PyDict_SetItem(dict, &_Py_ID(__doc__), value);
 }
 
+static PyObject *
+type_get_annotate(PyTypeObject *type, void *Py_UNUSED(ignored))
+{
+    if (!(type->tp_flags & Py_TPFLAGS_HEAPTYPE)) {
+        PyErr_Format(PyExc_AttributeError, "type object '%s' has no attribute '__annotate__'", type->tp_name);
+        return NULL;
+    }
+
+    PyObject *annotate;
+    PyObject *dict = lookup_tp_dict(type);
+    if (PyDict_GetItemRef(dict, &_Py_ID(__annotate__), &annotate) < 0) {
+        return NULL;
+    }
+    if (annotate) {
+        descrgetfunc get = Py_TYPE(annotate)->tp_descr_get;
+        if (get) {
+            Py_SETREF(annotate, get(annotate, NULL, (PyObject *)type));
+        }
+    }
+    else {
+        annotate = Py_None;
+        int result = PyDict_SetItem(dict, &_Py_ID(__annotate__), annotate);
+        if (result == 0) {
+            PyType_Modified(type);
```

Comment:
It's not clear to me that this should call `PyType_Modified`, since semantically "not set" and "set to None" are the same (the distinction shouldn't be observable), we just set the actual `None` lazily.

It's especially odd to me given that we don't currently call `PyType_Modified` below in `type_set_annotate`.

Reply:
We discussed this in person and it's probably fine to call PyType_Modified here.